### PR TITLE
Updated model

### PR DESCRIPTION
**Pull request by siddhant230**
updated the entire flow
created a deepface custom lib
created a face enhancement method

### CODE REVIEW

File: lib/deepface_custom/Face_extract.py [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+                (startX, startY) = (max(0, startX), max(0, startY))
+                (endX, endY) = (min(w - 1, endX), min(h - 1, endY))
+                face = frame[startY:endY, startX:endX]
+                try:
+                    face = cv2.resize(face, (224, 224))
+                    face = img_to_array(face)
+                    face = preprocess_input(face)
+                    face = np.expand_dims(face, axis=0)
+                except:
+                    pass
```
[the remark this answers is not in the document]
Comment (siddhant230) [reply lib/deepface_custom/Face_extract.py]: resolved

File: lib/deepface_custom/Face_extract.py [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+                faces.append(face)
+                locs.append((startX, startY, endX, endY))
+
+        # only make a predictions if at least one face was detected
+        if len(faces) > 0:
+            sx, sy, ex, ey = locs[-1]
+
+            return frame[sy:ey, sx:ex]
+        else:
+            return None
```
[the remark this answers is not in the document]
Comment (siddhant230) [reply lib/deepface_custom/Face_extract.py]: resolved

**Review by siddhant01 — APPROVED** [review]

File: src/deepface_similarity.py [hunk 3]
```diff
@@ -70,26 +77,23 @@ def find_similarity(self, image_1: np.array, image_2: np.array):
...
-                    return 'Different'
+                    return 'Different'"""
             else:
-                cv2.imshow('im1', face_1)
-                cv2.imshow('im2', face_2)
-                cv2.waitKey(0)
                 return "Too much Blurred; cannot find face"
         except:
-            print("Cannot find face")
+            logger.info("[ERROR] Cannot find face")
```
Comment (kritikam15) [inline src/deepface_similarity.py:95]: logger.error

File: src/deepface_similarity.py [hunk 4]
```diff
@@ -102,6 +106,9 @@ def match_faces(self, image_1_url: str, image_2_url: str):
...
         corrected_image1, corrected_image2 = self.process(image_1), image_2.copy()
-        result = self.find_similarity(corrected_image1, corrected_image2)
-
-        return result
+        if len(corrected_image1) > 0:
+            result = self.find_similarity(corrected_image1, corrected_image2)
+            logger.info("[INFO] Face comparison done!")
+            return result
+        else:
+            logger.info('[ERROR] Unable to extract and localize document!')
```
Comment (kritikam15) [inline src/deepface_similarity.py:114]: logger.error if an error

File: src/processing.py [hunk 5]
```diff
@@ -1,19 +1,26 @@
...
+console = logging.StreamHandler()
+logger.addHandler(console)
 
 class Process:
 
     def __init__(self, model):
         try:
             self.model = model
-            print('Model loaded successfully')
+            logger.info("[INFO] Localization Model Loaded")
```
Comment (kritikam15) [inline src/processing.py:18]: no need to write `Info`, its logger.info which does this work

File: src/deepface_similarity.py (hunk 3 above)
Comment (siddhant01) [reply src/deepface_similarity.py:87]: rectified

File: src/processing.py (hunk 5 above)
Comment (siddhant01) [reply src/processing.py:18]: retified

File: src/deepface_similarity.py (hunk 4 above)
Comment (siddhant01) [reply src/deepface_similarity.py]: rectified

**Review by kritikam15 — APPROVED** [review]

**Pull request merged**